sdk/include/opentelemetry/sdk/instrumentationscope/instrumentation_scope.h

```diff
@@ -158,6 +159,8 @@ class InstrumentationScope
   InstrumentationScopeAttributes attributes_;
 };
 
+template <class T>
+using ScopeConfigurator = std::function<T(const InstrumentationScope &)>;
```

Comment:
A couple of comments.

(1) 

ScopeConfigurator<T> should be an abstract class, with a virtual configure() method.

This will help to define different kinds of configurators later, including the various builtins helpers.

(2)

One of the configurators is the always enabled one, so implement a AlwaysOnScopeConfigurator sub class.

Have a static AlwaysOnScopeConfigurator::Create() method, to return an instance.

(3)

Likewise, AlwaysOffScopeConfigurator::Create()

(4)

All this should be in dedicated headers, so instrumentation_scope.h should be unchanged.

Reply:
Sounds good, I had a class before, but removed since it was easily replaceable by a function. I was expecting to put the built-in helpers in a static utility class, but this approach might look better.

I will give this a try.

Reply:
In the mean time, the spec configuration repository is defining the yaml bits for this:

* https://github.com/open-telemetry/opentelemetry-configuration/pull/140

Instead of "AlwaysOn" / "AlwaysOff", the following is probably better:

class ScopeConfigurator;
class DefaultScopeConfigurator : public ScopeConfigurator;

and later

class NamedScopeConfigurator : public ScopeConfigurator; (for name matching)

Reply:
My one concern with this approach is that it would require us to create and maintain multiple classes which wont necessary hold state and could therefore also be replaced by functions.

Based on these suggestions, I have created a Builder based approach that allows for helper functions to add functionality to a scope configurator. The existing tests show how this change would look like from the user's perspective.

Please take a look and let me know your thoughts on that.